## Data Loading

In [49]:
# Load the dataset and preview the first rows
import pandas as pd
df = pd.read_csv('Sample-Superstore.csv', encoding='latin1')
df.head()

,Row ID,Order ID,Order Date,Ship Date,Ship Mode,Customer ID,Customer Name,Segment,Country,City,...,Postal Code,Region,Product ID,Category,Sub-Category,Product Name,Sales,Quantity,Discount,Profit
0,1,CA-2016-152156,11/8/2016,11/11/2016,Second Class,CG-12520,Claire Gute,Consumer,United States,Henderson,...,42420,South,FUR-BO-10001798,Furniture,Bookcases,Bush Somerset Collection Bookcase,261.9600,2,0.00,41.9136
1,2,CA-2016-152156,11/8/2016,11/11/2016,Second Class,CG-12520,Claire Gute,Consumer,United States,Henderson,...,42420,South,FUR-CH-10000454,Furniture,Chairs,"Hon Deluxe Fabric Upholstered Stacking Chairs,...",731.9400,3,0.00,219.5820
2,3,CA-2016-138688,6/12/2016,6/16/2016,Second Class,DV-13045,Darrin Van Huff,Corporate,United States,Los Angeles,...,90036,West,OFF-LA-10000240,Office Supplies,Labels,Self-Adhesive Address Labels for Typewriters b...,14.6200,2,0.00,6.8714
3,4,US-2015-108966,10/11/2015,10/18/2015,Standard Class,SO-20335,Sean O'Donnell,Consumer,United States,Fort Lauderdale,...,33311,South,FUR-TA-10000577,Furniture,Tables,Bretford CR4500 Series Slim Rectangular Table,957.5775,5,0.45,-383.0310
4,5,US-2015-108966,10/11/2015,10/18/2015,Standard Class,SO-20335,Sean O'Donnell,Consumer,United States,Fort Lauderdale,...,33311,South,OFF-ST-10000760,Office Supplies,Storage,Eldon Fold 'N Roll Cart System,22.3680,2,0.20,2.5164


In [ ]:
# Check the number of rows and columns
df.shape

(9994, 21)

## Data Cleaning

In [ ]:
# Inspect column types and check for missing values
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9994 entries, 0 to 9993
Data columns (total 21 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Row ID         9994 non-null   int64  
 1   Order ID       9994 non-null   object 
 2   Order Date     9994 non-null   object 
 3   Ship Date      9994 non-null   object 
 4   Ship Mode      9994 non-null   object 
 5   Customer ID    9994 non-null   object 
 6   Customer Name  9994 non-null   object 
 7   Segment        9994 non-null   object 
 8   Country        9994 non-null   object 
 9   City           9994 non-null   object 
 10  State          9994 non-null   object 
 11  Postal Code    9994 non-null   int64  
 12  Region         9994 non-null   object 
 13  Product ID     9994 non-null   object 
 14  Category       9994 non-null   object 
 15  Sub-Category   9994 non-null   object 
 16  Product Name   9994 non-null   object 
 17  Sales          9994 non-null   float64
 18  Quantity

In [ ]:
# Check for missing values in each column
df.isnull().sum()

Row ID           0
Order ID         0
Order Date       0
Ship Date        0
Ship Mode        0
Customer ID      0
Customer Name    0
Segment          0
Country          0
City             0
State            0
Postal Code      0
Region           0
Product ID       0
Category         0
Sub-Category     0
Product Name     0
Sales            0
Quantity         0
Discount         0
Profit           0
dtype: int64

In [ ]:
# Check for duplicate rows
df.duplicated().sum()

np.int64(0)

In [ ]:
# Convert date columns to proper datetime format for time-based analysis
df['Order Date'] = pd.to_datetime(df['Order Date'])
df['Ship Date'] = pd.to_datetime(df['Ship Date'])
df.dtypes

Row ID                    int64
Order ID                 object
Order Date       datetime64[ns]
Ship Date        datetime64[ns]
Ship Mode                object
Customer ID              object
Customer Name            object
Segment                  object
Country                  object
City                     object
State                    object
Postal Code               int64
Region                   object
Product ID               object
Category                 object
Sub-Category             object
Product Name             object
Sales                   float64
Quantity                  int64
Discount                float64
Profit                  float64
dtype: object

## 1. Category & Sub-Category Profitability Analysis

In [ ]:
# Calculate total sales and profit per category, then add a profit margin column
category_summary = df.groupby('Category')[['Sales', 'Profit']].sum()
category_summary['Profit Margin'] = category_summary['Profit'] / category_summary['Sales']
category_summary

,Sales,Profit,Profit Margin
Category,,,
Furniture,741999.7953,18451.2728,0.024867
Office Supplies,719047.0320,122490.8008,0.170352
Technology,836154.0330,145454.9481,0.173957


In [ ]:
# Calculate total sales and profit per sub-category, add profit margin, sort by margin
subcategory_summary = df.groupby('Sub-Category')[['Sales', 'Profit']].sum()
subcategory_summary['Profit Margin'] = subcategory_summary['Profit'] / subcategory_summary['Sales']
subcategory_summary.sort_values('Profit Margin')

,Sales,Profit,Profit Margin
Sub-Category,,,
Tables,206965.5320,-17725.4811,-0.085645
Bookcases,114879.9963,-3472.5560,-0.030228
Supplies,46673.5380,-1189.0995,-0.025477
Machines,189238.6310,3384.7569,0.017886
Chairs,328449.1030,26590.1663,0.080957
Storage,223843.6080,21278.8264,0.095061
Phones,330007.0540,44515.7306,0.134893
Furnishings,91705.1640,13059.1436,0.142404
Binders,203412.7330,30221.7633,0.148574


**Finding:** Furniture has a much lower profit margin (2.5%) compared to 
Office Supplies and Technology (both ~17%), despite similar sales volume. 
Digging into sub-categories reveals the cause: Tables (-8.6% margin) and 
Bookcases (-3.0% margin) are actually losing money, dragging down the 
entire Furniture category.


## 2. Regional & State Profit Analysis

In [ ]:
# Calculate total sales and profit per region, add profit margin, sort by profit
region_summary = df.groupby('Region')[['Sales', 'Profit']].sum()
region_summary['Profit Margin'] = region_summary['Profit'] / region_summary['Sales']
region_summary.sort_values('Profit')

,Sales,Profit,Profit Margin
Region,,,
Central,501239.8908,39706.3625,0.079216
South,391721.9050,46749.4303,0.119343
East,678781.2400,91522.7800,0.134834
West,725457.8245,108418.4489,0.149448


In [62]:
# Calculate total sales and profit per state, add profit margin, sort by profit
state_summary = df.groupby('State')[['Sales', 'Profit']].sum()
state_summary['Profit Margin'] = state_summary['Profit'] / state_summary['Sales']
state_summary.sort_values('Profit').head(10)

,Sales,Profit,Profit Margin
State,,,
Texas,170188.0458,-25729.3563,-0.151182
Ohio,78258.1360,-16971.3766,-0.216864
Pennsylvania,116511.9140,-15559.9603,-0.133548
Illinois,80166.1010,-12607.8870,-0.157272
North Carolina,55603.1640,-7490.9122,-0.134721
Colorado,32108.1180,-6527.8579,-0.203309
Tennessee,30661.8730,-5341.6936,-0.174213
Arizona,35282.0010,-3427.9246,-0.097158
Florida,89473.7080,-3399.3017,-0.037992


In [63]:
# Calculate total sales and profit per city (grouped with state to avoid mixing same-name cities), showing the 10 worst-performing cities
city_summary = df.groupby(['State', 'City'])[['Sales', 'Profit']].sum()
city_summary['Profit Margin'] = city_summary['Profit'] / city_summary['Sales']
city_summary.sort_values('Profit').head(10)

Sales      Profit  Profit Margin
State          City                                                
Pennsylvania   Philadelphia  109077.0130 -13837.7674      -0.126862
Texas          Houston        64504.7604 -10153.5485      -0.157408
               San Antonio    21843.5280  -7299.0502      -0.334152
Ohio           Lancaster       8202.6250  -7149.6180      -0.871626
Illinois       Chicago        48539.5410  -6654.5688      -0.137096
North Carolina Burlington     12681.2820  -5894.5269      -0.464821
Colorado       Louisville      5070.4160  -3406.2095      -0.671781
Texas          Dallas         20131.9322  -2846.5257      -0.141394
Arizona        Phoenix        11000.2570  -2790.8832      -0.253711
Florida        Jacksonville   39133.3280  -2445.6608      -0.062496

**Finding:** Texas has the largest absolute loss (-$25,729), while Ohio has 
the worst relative performance (-21.7% profit margin). Drilling down to the 
city level reveals that Ohio's poor performance is driven almost entirely by 
one city, Lancaster, which has an extreme -87.2% profit margin — showing 
that state-level losses can be concentrated in just a few locations rather 
than spread evenly.


## 3. Discount vs Profitability Analysis

In [66]:
# Calculate average discount and profit for each sub-category, sorted by discount (highest first)
discount_analysis = df.groupby('Sub-Category')[['Discount', 'Profit']].mean()
discount_analysis.sort_values('Discount', ascending=False)

,Discount,Profit
Sub-Category,,
Binders,0.372292,19.843574
Machines,0.306087,29.432669
Tables,0.261285,-55.565771
Bookcases,0.211140,-15.230509
Chairs,0.170178,43.095894
Appliances,0.166524,38.922758
Copiers,0.161765,817.909190
Phones,0.154556,50.073938
Furnishings,0.138349,13.645918



**Finding:** Tables and Bookcases receive high average discounts (26% and 
21%) and show negative average profit, confirming that heavy discounting 
hurts profitability for these sub-categories. However, this relationship 
isn't strictly linear — Binders receives the highest discount (37%) but 
remains profitable, suggesting the impact of discounting depends on each 
product's baseline profit margin.
